src/util/logging.py

```diff
@@ -17,30 +17,54 @@
 
 from util.context import Context
 
+LOG_THROTTLING_LIMIT_PER_CALLER = 10
+
+log_call_count = dict()
+
+
+def log_multiline_message(message: Text, caller: Text):
+    # display logs called from one spot no more than the specified amount of times
+    if check_if_caller_exceeded_limit(caller):
+        return
 
-def log_multiline_message(message: Text):
     # need to modify endline char to have multiline log record not split into multiple log entries in CloudWatch:
     message = message.replace('\n', ' ')
     print(message)
 
 
-def debug_log_multiline_message(message: Text, context: Context):
+def check_if_caller_exceeded_limit(caller):
+    log_calls_left = LOG_THROTTLING_LIMIT_PER_CALLER - log_call_count.get(caller, 0)
+
+    if log_calls_left == 0:
+        log_call_count[caller] = log_call_count.get(caller, 0) + 1
+        print(f"Logging calls from caller '{caller}' exceeded the throttling limit of",
+              f"{LOG_THROTTLING_LIMIT_PER_CALLER}. Further logs from this caller will be discarded")
+
+    caller_exceeded_limit = log_calls_left <= 0
+    if not caller_exceeded_limit:
+        log_call_count[caller] = log_call_count.get(caller, 0) + 1
```

Comment:
It looks much better, thanks.

For log_calls_left == 0 though, aren't you bumping up the number twice (39 & 46)? That would still work but seems wrong. I think I'd just remove 39 and it would be nice and clean.
`if log_calls_left == 0:` > true

`caller_exceeded_limit = log_calls_left <= 0
    if not caller_exceeded_limit:` > true

Reply:
Unless I am missing something, when log_calls_left == 0, it goes into the first if, but caller_exceeded_limit evaluates to True and doesn't go into the second if

Reply:
Looked at it too long, I missed `not` xD
Sorry, good to go:P